contracts/MarginCalculator.sol

```diff
@@ -16,19 +17,20 @@ import {MarginVault} from "./libs/MarginVault.sol";
  * @author Opyn
  * @notice Calculator module that checks if a given vault is valid, calculates margin requirements, and settlement proceeds
  */
-contract MarginCalculator {
+contract MarginCalculator is Ownable {
     using SafeMath for uint256;
     using FPI for FPI.FixedPointInt;
 
-    /// @dev oracle module
-    OracleInterface public oracle;
+    /// @dev decimals option upper bound value, spot shock and oracle deviation
+    uint256 internal constant SCALING_FACTOR = 27;
 
     /// @dev decimals used by strike price and oracle price
     uint256 internal constant BASE = 8;
 
-    /// @dev FixedPoint 0
-    FPI.FixedPointInt internal ZERO = FPI.fromScaledUint(0, BASE);
+    /// @notice auction length
+    uint256 public constant AUCTION_TIME = 3600;
```

Comment:
I lean towards this being adjustable.  If we start having failed liquidations we may need to adjust this quickly.

I guess we could always deploy a new controller, set all the values and then migrate it in the address book.  That would require a bunch of tx, but only one tx to migrate, so we don't have a broken/controller with non finalized values.

Reply:
If we do allow it to be set, we could set a min value - admin can prb steal a lot of margined funds by atomically:

-flashloan bunch of eth
-changing margin requirements
-setting auction_time to 0
-liquidating a bunch of folks (they might need a recursive operate function to do this have to mint+liquidate+burn)
-setting margin req back to normal + auction time back to normal

Reply:
To update this, we need to redeploy `MarginCalculator` not the `Controller`, and yeah it will require more txs but I really prefer keeping this constant, as it is more gas efficient, and still if we have failed liquidation w can still batch deployment and params config in one transaction.